titan-server/src/accounts/routes.rs

```diff
@@ -41,85 +43,59 @@ pub fn woltlab_login(
     titan_primary: TitanPrimary,
     login_creds: Json<WoltlabLoginRequest>,
     app_config: State<config::AppConfig>,
-) -> Result<Json<WoltlabLoginResponse>, status::NotFound<String>> {
+) -> ApiResponse<WoltlabLoginResponse> {
     let wcf_db = &*unkso_main;
     let titan_db = &*titan_primary;
+    let res = users::wcf_find_by_user_id(login_creds.user_id, wcf_db)
+        .map_err(ApiError::from)
+        .and_then(|wcf_user| {
+            let is_valid = woltlab_auth_helper::check_password(
+                &wcf_user.password, &login_creds.cookie_password);
+            match is_valid {
```

Comment:
This is probably better written as an if/else instead of a match since it's operating off of boolean values.

i.e.

```rust
if is_valid {
    Ok(wcf_user)
} else {
    Err(ApiError::AuthenticationError)
}
```